BUILD.bazel

```diff
@@ -882,30 +911,41 @@ cc_library(
     ],
 )
 
-cc_library(
-    name = "torch_python",
+cc_binary(
+    name = "libtorch_python.so",
     srcs = libtorch_python_core_sources
-        + if_cuda(libtorch_python_cuda_sources)
+        + if_cuda(glob(libtorch_python_cuda_sources, exclude = ["torch/csrc/cuda/python_nccl.cpp"]))
         + if_cuda(libtorch_python_distributed_sources)
-        + GENERATED_AUTOGRAD_PYTHON,
-    hdrs = glob([
-        "torch/csrc/generic/*.cpp",
+        + GENERATED_AUTOGRAD_PYTHON
+        + glob([
+            "torch/csrc/generic/*.cpp",
     ]),
     copts = COMMON_COPTS + if_cuda(["-DUSE_CUDA=1"]),
     deps = [
-        ":torch",
-        ":shm",
+        ":torch_headers",
         "@pybind11",
+        "@kineto",
+        "@fmt",
+    ] + if_cuda(["@cudnn", "@cuda"]),
+    linkshared = True,
+)
+
+cc_library(
+    name = "torch_python",
+    srcs = [
+        ":libtorch_python.so",
     ],
 )
 
-pybind_extension(
-    name = "torch/_C",
+cc_binary(
+    name = "torch/_C.so",
     srcs = ["torch/csrc/stub.c"],
     deps = [
         ":torch_python",
-        ":aten_nvrtc",
```

Comment:
We dont need to link `aten_nvrtc` directly since bazel build is without `-DUSE_DIRECT_NVRTC`.